### ISSUE TITLE
how to disable segment background (make it transparent) in colorscheme?

### ISSUE THREAD

**Issue by Naitreey** [powerline/powerline#1887]
I'm trying to create a powerline shell prompt. I have achieved a prompt consisting of two lines, where the first line is various powerline segments, and the second line is a bare shell prompt like so:
![2018-02-26-020213_3440x1440_scrot](https://user-images.githubusercontent.com/10528297/36644701-447d82c8-1a99-11e8-8cab-2e050b9ac475.png)
The problem is I don't know what `bg` color value to use to make the `$` segment transparent, i.e.
use whatever color the terminal color scheme provide.

Am I missing something obvious?

**Comment by ZyX-I** [comment]
You will need to create a color in colorscheme which is the same as terminal background. Powerline *always* sets both background and foreground colors to something.

**Comment by Naitreey** [comment]
Okay, I see. Thanks.

**Comment by nxuv** [comment]
For anyone still wondering about this:
```python
    # RESET is not a real color code. It is used as in indicator
    # within the code that any foreground / background color should
    # be cleared
    RESET = -1
```
So if you set color to `-1` it'll dissapear. I tested it, it works

**Comment by ttelford** [comment]
> For anyone still wondering about this:
> 
> ```python
>     # RESET is not a real color code. It is used as in indicator
>     # within the code that any foreground / background color should
>     # be cleared
>     RESET = -1
> ```
> 
> So if you set color to `-1` it'll dissapear. I tested it, it works

This is mostly in case anybody else runs into [XKCD 979](https://xkcd.com/979/) like I did:
![XKCD 979](https://imgs.xkcd.com/comics/wisdom_of_the_ancients.png)


Are you sure that that trick works for *this* implementation of Powerline, and not a different version?

I ask because the `RESET = -1` is _not_ JSON syntax, and there's nothing similar to it in the source code. For that matter, if I set a color in the most obvious place (`~/.config/powerline/colors.json`), and then use the `reset` color as claimed:

```json
  {
      "colors": {
          "reset": -1,
          "black": 16,
  }
```

`~/.config/powerline/colorschemes/default.json`
```json
{
      "name": "Default",
      "groups": {
...
          "background":                 { "fg": "reset", "bg": "reset", "attrs": [] },
...
}
```

The powerline log simply throws a stack trace in `set_segment_highlighting` that it can't set that color. (And changing the "reset" name to any other color then succeeds.)

**Comment by sghng** [comment]
also looking for a solution to set a color to transparent. the closes thing I’ve found is using “set global-options status-style bg=default” ( I forgot the exact syntax)

**Comment by PH111P** [comment]
Transparency as a color is not currently supported. I am not sure where @al1-ce got the quoted snippet from, but it is not from this repository, I think.

Let me briefly outline what would need to be changed.
- The code that parses and constructs colors needs to allow for "transparency". One possible option is to allow specifying ARGG hex values for colors. (I did implement something like that in my own fork some years ago. Most bindings for the terminal didn't seem to work well with the alpha channel, though.)
- Each renderer needs to be adapted to allow for transparency/new color formats. For some bindings this is more involved than for others; IIRC for e.g. vim it seemed to be very difficult. (In my fork, I did adapt the renderer only for the lemonbar binding; hence I did not merge it into this repository.)
- Tests and documentation need to be updated accordingly.

Concluding, there is little hope that you will be able to easily set transparency as a bg color in a terminal soon. (However, I didn't look into this issue for a couple of years now, so things might have changed in the meantime, making transparency somewhat easier to implement.)
